### DISCUSSION TITLE
Can supabase be used server-side?

### DISCUSSION THREAD

**Discussion by benallfree** [supabase/supabase#5591] [Questions]
If I attempt to use supabase in a remix loader, I get 

```
    xhr = new globalThis.XMLHttpRequest();
          ^

TypeError: globalThis.XMLHttpRequest is not a constructor
```

Does anyone know if supabase can be used server-side?

For a little more context, here's my remix loader:

```

export const action: ActionFunction = async ({ request, params }) => {
  const res = await supabase.auth.signIn({ email: 'foo@bar.com' })
  console.log({ res })
}
```

**Comment by danieldietzel** [comment]
I'm also wondering this. Functions that work fine on the client side return undefined on the server / loader side in remix.

**Reply by GaryAustin1** [reply]
Server side you have to set persistSession:false as there is no local storage.  If you don't getSession() and other things will null the in memory session. 

Also see this https://supabase.com/docs/guides/auth/server-side-rendering

**Comment by astonfuture** [comment]
I would look into using the supabase remix auth helpers package and follow the video guides from from @dijonmusters on setting up supabase + remix:

https://youtu.be/Viaed7XWCY8?si=_0d6XPyUoBSyZQgi

Note that this guide uses the implicit flow for auth rather than the pkce flow which I believe is a more recent feature and the currently recommended flow.
